### PR TITLE
v0.6

### PR DESCRIPTION
**Pull request by NotAShelf**
Continuation of #197. Rebased around v0.6 branch to avoid merge conflicts.

# TODO 

- neo-tree
- Move dashboard plugins to the new setupOpts format
- Re-visit tabline, lualine and scrollbar configurations
- Base16 theming

## Things I might want to do:

- Other suggestions by contributors and maintainers alike
- Look into lazy loading via Lazy (CC: @horriblename)

# Done

- Switch to makeNeovimConfig, or move to "unstable" neovim wrapper
- rebase `fidget.nvim` around the latest master (previously legacy branch) by FrothyMarrow
- always include lua parser with treesitter by FrothyMarrow
- allow configurable category labels with which-key by FrothyMarrow
- add image.nvim by NotAShelf
- move home-manager module to a directory dedicated to modules, accommodating for the possibilities of NixOS and Darwin modules in the future. by NotAShelf
   - `lib/modules/default.nix` -> `flake/modules/hm-module.nix`
- Clean up the initial root directory of the project.
   - Files that belong in their respective directories (e.g. assets) to be moved
- Reduce dependency cycles and layered imports in `modules/`
   - All plugins to be moved to `modules/plugins/`
   - Option definitions separated from configurations
   - `Basic` and `Core` to be isolated as `modules/core` and `modules/neovim` (previously basic)`
     - `core` will contain wrapper and build related files
     - `neovim` will contain the most basic vim/neovim options that are passed to base configurations.
- Cleans up and re-organizes lib so that files and functions are easier to search by name.
  - Revisit lib structure, overlay lib functions so that they are available on on top level (e.g. lib.toLuaObject as well as lib.lua.toLuaObject)
- Move neovim config builder/helper functions into lib, and call from lib where necessary. This has been an issue while implementinig the home manager module, and I'd like to avoid repeating the same mistake again.
- Make lib usage more explicit -> Instead of `inherit (lib) concatStringsSep`, we use `inherit (lib.strings) concatStringsSep` to ensure clarity. This slightly clutters the beginning of each file, but it's more understandable where exactly the function comes from (especially when working with the extended lib)

### CODE REVIEW

**Comment by NotAShelf** [comment]
@horriblename sorry to move around so much, I'll have to ask you to rebase #228, #229 and #230 around this PR again. This *should* be the final, but it also means that we're feature freezing `main` for the time being.

**Comment by justchokingaround** [comment]
modes-nvim causes visual lines selection to disappear
image

**Comment by justchokingaround** [comment]
also please add neo-tree-nvim

**Comment by NotAShelf** [comment]
neo-tree is definitely in my TODO, most likely after I merge #238

**Comment by justchokingaround** [comment]
can we get codeium as well pls (for assistants) :pleading_face: 
https://github.com/Exafunction/codeium.vim

**Comment by NotAShelf** [comment]
> can we get codeium as well pls (for assistants) 🥺 [Exafunction/codeium.vim](https://github.com/Exafunction/codeium.vim)

I'll add a module for it after I'm done with the refactor, but remember that you can add custom plugins easily using the [updated custom plugin API](https://notashelf.github.io/neovim-flake/index.xhtml#ch-custom-plugins)

File: docs/manual/hacking/guidelines.md [hunk 1]
```diff
@@ -1,71 +1,84 @@
...
-:::{.note}
-
-As of v0.5, neovim-flake is itself documented using full markdown in both module options and the manual. With
-v0.6, this manual has also been converted to markdown in full.
+Most, if not all, changes warrant changes to the documentation. Module options
+should be documented with [Nixpkgs-flavoured Markdown](https://nixos.org/manual/nixpkgs/unstable/#sec-contributing-markup),
+albeit with exceptions.
 
+::: {.note}
+As of **v0.5**, **vf** is itself documented using full markdown in both module
```
Comment (horriblename) [inline docs/manual/hacking/guidelines.md:21]: vf?

**Review by horriblename — CHANGES_REQUESTED** [review]

File: docs/manual/hacking/guidelines.md (hunk 1 above)
Comment (horriblename) [reply docs/manual/hacking/guidelines.md:21]:
```suggestion
As of **v0.5**, **nvf** is itself documented using full markdown in both module
```

Comment (NotAShelf) [reply docs/manual/hacking/guidelines.md]: I accidentally leaked the future future name...

File: modules/plugins/visuals/fidget/fidget.nix [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+          };
+          format_group_name = mkOption {
+            description = "How to format a progress notification group's name";
+            type = luaInline;
+            default = mkLuaInline ''
+              function(group) return tostring(group) end
+            '';
+          };
+          overrides = mkOption {
+            description = "Override options from the default notification config";
```
Comment (FrothyMarrow) [inline modules/plugins/visuals/fidget/fidget.nix:179]: Can we make notification a submodule type and reuse it here? Makes it less cursed

**Review by horriblename — DISMISSED** [review]
LGTM

**Pull request merged**